api/client.go

```diff
@@ -670,24 +709,41 @@ func (c *Client) SetBackoff(backoff retryablehttp.Backoff) {
 // Also, only the client's config is currently copied; this means items not in
 // the api.Config struct, such as policy override and wrapping function
 // behavior, must currently then be set as desired on the new client.
+//
+// One exception to this is the addr field of Client, which is also copied over.
+// Otherwise you have the confusing result of creating a client, customizing
+// the address via client.SetAddress() then cloning it and having it show the
+// default address again, instead of the one you just set. I'm not even sure
+// why the client has addr as a member at all, other than the convenience of
+// having it pre-parsed into a URL instead of a string.
 func (c *Client) Clone() (*Client, error) {
 	c.modifyLock.RLock()
-	c.config.modifyLock.RLock()
 	config := c.config
 	c.modifyLock.RUnlock()
 
+	config.modifyLock.RLock()
 	newConfig := &Config{
-		Address:    config.Address,
-		HttpClient: config.HttpClient,
-		MaxRetries: config.MaxRetries,
-		Timeout:    config.Timeout,
-		Backoff:    config.Backoff,
-		CheckRetry: config.CheckRetry,
-		Limiter:    config.Limiter,
+		Address:          config.Address,
+		HttpClient:       config.HttpClient,
+		MaxRetries:       config.MaxRetries,
+		Timeout:          config.Timeout,
+		Backoff:          config.Backoff,
+		CheckRetry:       config.CheckRetry,
+		Limiter:          config.Limiter,
+		OutputCurlString: config.OutputCurlString,
 	}
 	config.modifyLock.RUnlock()
+	client, err := NewClient(newConfig)
+	if err != nil {
+		return nil, err
+	}
 
-	return NewClient(newConfig)
+	err = client.SetAddress(newConfig.Address)
```

Comment:
Looking at the code some more I'm confused why we need to call SetAddress here, since NewClient does the equivalent.

Reply:
I'll double check this to be sure, but my memory is that either `client.addr` or `config.Address` ended up not being set correctly without this line.